### PR TITLE
chore(nargo): report bubbled up errors using eyre

### LINKED ISSUE #686: nargo colours subsequent cli lines red after encountering an error
# Description

It appears nargo colours the subsequent lines of the cli red after having an error

## Aim

change this to nornal behaviour

## Expected behavior

It should only colour the line `error: aborting due to {error_count} previous errors` red i think.

## Bug

# To reproduce

1. create an error in a source file
2. run `nargo build`
3. check rest of the cli lines
4.

# Environment

nargo 0.1.0 on mac m1

- [x] Using `nargo`:

  - [x] Binary: 
  - [ ] Compiled from source: 

- [ ] Using TypeScript:

  - @noir-lang/noir_wasm: 
  - @noir-lang/barretenberg: 
  - @noir-lang/aztec_backend: 

# Additional context

after an error and run `nargo --help`

### LINKED ISSUE #819: Replace `thiserror` in `nargo` with `anyhow`
# Problem

It seems `thiserror` is usually used in libraries and `anyhow` is used more in binaries. 

Since `nargo` is a binary, it might be better for it to adopt `anyhow` for errors

# Solution

replace `thiserror` in `nargo` with `anyhow`

# Alternatives considered

alternatively, we can keep using `thiserror` for `nargo`

# Additional context

### PR DESCRIPTION
**Pull request by TomAFrench**
# Related issue(s)

Resolves #819 
Resolves #686 

# Description

## Summary of changes

Rather than explicitly writing errors to the console and then calling `std::process::exit(1)` we now wrap the bubbled up error in an `anyhow::Result` and just return it from `main`.

## Dependency additions / changes

Added `anyhow` as dep of nargo.

## Test additions / changes

# Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [x] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [x] I have reviewed the changes on GitHub, line by line.
- [x] I have ensured all changes are covered in the description.
- [ ] This PR requires documentation updates when merged.

# Additional context

### CODE REVIEW

**Comment by TomAFrench** [comment]
Ah, thanks yes it does (as we no longer colour the terminal)

**Review by jfecher — COMMENTED** [review]
I'm not sure if it's worth it to add a new dependency just to avoid writing:
```
if let Err(err) = result {
    err.write()
}
```

**Comment by phated** [comment]
I usually agree with @jfecher about adding dependencies but I think getting anyhow *into* the project would be helpful for the future. It's one of the (surprisingly) few crates that I've seen recommended by everyone (probably just thiserror and anyhow!)

**Comment by TomAFrench** [comment]
Actually, it looks like our existing dependency of `color_eyre` can perform much the same role as `anyhow` would here.

File: crates/nargo/src/errors.rs [hunk 1]
```diff
@@ -35,18 +34,6 @@ impl From<OpcodeResolutionError> for CliError {
...
 
-impl CliError {
-    pub(crate) fn write(&self) -> ! {
-        let mut stderr = StandardStream::stderr(ColorChoice::Always);
-        stderr
-            .set_color(ColorSpec::new().set_fg(Some(Color::Red)))
-            .expect("cannot set color for stderr in StandardStream");
-        writeln!(&mut stderr, "{self}").expect("cannot write to stderr");
-
-        std::process::exit(1)
```
Comment (kevaundray) [inline crates/nargo/src/errors.rs:46]: one less place with `std::process::exit(1)` :D

**Review by kevaundray — DISMISSED** [review]

**Review by jfecher — APPROVED** [review]

**Pull request merged**